index.js

```diff
@@ -309,11 +309,9 @@ function additionalProperty (location) {
 
     return code
   }
-  let apLocation = mergeLocation(location, 'additionalProperties')
-  if (ap.$ref) {
-    apLocation = resolveRef(location, ap.$ref)
-    ap = apLocation.schema
-  }
+  const apLocation = ap.$ref
+    ? resolveRef(location, ap.$ref)
+    : mergeLocation(location, 'additionalProperties')
```

Comment:
There is a bug here. We should always call `mergeLocation` here and pass `apLocation` to the `resolveRef` instead of `location`.

Reply:
I dont quite understand your instructions:
I just saw that alert and wanted to fix it. If there is a bug, I would actually need a test to know that I am actually fixing something. Currently all the tests pass.

Probably it is the best to close this PR and open a bug issue.

Reply:
It is not really the bug. Since resolveRef use only `schemaId` property of the `location` object, it will not fail, because `location` and `apLocation` have the same `schemaId`. It is just a proper way to do this. There is no test needed.